segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```diff
@@ -73,6 +77,31 @@ public void tearDown() throws Exception {
         s3Proxy.stop();
     }
 
+    //region If-none-match test
+    /**
+     * Tests the create() method with if-none-match set.
+     */
+    @Test
+    public void testCreateIfNoneMatch() {
+        adapterConfig = ExtendedS3StorageConfig.builder()
+                                               .with(ExtendedS3StorageConfig.BUCKET, adapterConfig.getBucket())
+                                               .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                               .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                               .with(ExtendedS3StorageConfig.ROOT, "test")
+                                               .with(ExtendedS3StorageConfig.URI, endpoint)
+                                               .with(ExtendedS3StorageConfig.IFNONEMATCHWORKS, true)
+                                               .build();
+        String segmentName = "foo_open";
+        try (Storage s = createStorage()) {
+            s.initialize(DEFAULT_EPOCH);
+            s.create(segmentName, null).join();
+            assertThrows("create() did not throw for existing StreamSegment.",
```

Comment:
The test code just executes the `putObject` with the setting enabled. Our test implementations do not behave same as the default extended S3 storage in this case. It does not exercise the part where exception is thrown because of `if-none-match` check.